apps/dashboard/app/javascript/packs/files/data_table.js

```diff
@@ -0,0 +1,441 @@
+import 'datatables.net';
+import 'datatables.net-bs4/js/dataTables.bootstrap4';
+import 'datatables.net-select';
+import 'datatables.net-select-bs4';
+import Handlebars from 'handlebars';
+import {EVENTNAME as CLIPBOARD_EVENTNAME} from './clip_board.js';
+import {EVENTNAME as FILEOPS_EVENTNAME} from './file_ops.js';
+import {EVENTNAME as SWAL_EVENTNAME} from './sweet_alert.js';
+
+export { CONTENTID, EVENTNAME };
+
+const EVENTNAME = {
+    getJsonResponse: 'getJsonResponse',
+    reloadTable: 'reloadTable',
+    goto: 'goto',
+};
+
+const CONTENTID = {
+    table: '#directory-contents',
+};
+
+let table = null;
+
+jQuery(function () {
+    table = new DataTable();
+
+    /* BUTTON ACTIONS */
+    $("#new-file-btn").on("click", function () {
+        $(CONTENTID.table).trigger(FILEOPS_EVENTNAME.newFilePrompt);
+    });
+
+    $("#new-folder-btn").on("click", function () {
+        $(CONTENTID.table).trigger(FILEOPS_EVENTNAME.newFolderPrompt);
+    });
+
+    $("#download-btn").on("click", function () {
+        let selection = table.getTable().rows({ selected: true }).data();
+        const eventData = {
+            selection: selection
+        };
+
+        $(CONTENTID.table).trigger(FILEOPS_EVENTNAME.download, eventData);
+
+    });
+
+    $("#delete-btn").on("click", function () {
+        let files = table.getTable().rows({ selected: true }).data().toArray().map((f) => f.name);
+        const eventData = {
+            files: files
+        };
+
+        $(CONTENTID.table).trigger(FILEOPS_EVENTNAME.deletePrompt, eventData);
+
+    });
+
+    $("#copy-move-btn").on("click", function () {
+        let selection = table.getTable().rows({ selected: true }).data();
+        const eventData = {
+            selection: selection
+        };
+
+        $(CONTENTID.table).trigger(CLIPBOARD_EVENTNAME.updateClipboard, eventData);
+
+    });
```

Comment:
Why does this still import here? I thought data_table wouldn't import _anything_ only export the event's that _it's_ firing. 

You've just tied to the knowledge that the `copy-move-btn` directly relates to updating the clipboard, which it does, but datatable shouldn't know that.  

```suggestion
    $("#copy-move-btn").on("click", function () {
        let selection = table.getTable().rows({ selected: true }).data();
        const eventData = {
            selection: selection
        };

        $(CONTENTID.table).trigger(EVENTNAME.copyMove, eventData);

    });
```

Reply:
Good catch!